bouncer/views.py

```diff
@@ -25,11 +29,16 @@ def annotation(self):
                 id=self.request.matchdict["id"])
         except exceptions.NotFoundError:
             statsd.incr("views.annotation.404.annotation_not_found")
-            raise httpexceptions.HTTPNotFound()
+            raise httpexceptions.HTTPNotFound("Annotation not found")
 
         annotation_id = document["_id"]
         document_uri = document["_source"]["uri"]
 
+        if not (document_uri.startswith("http://") or
+                document_uri.startswith("https://")):
+            raise httpexceptions.HTTPUnprocessableEntity(
+                _("We can only link to http(s) URLs"))
```

Comment:
Perhaps this is even clearer?

"Sorry, but it looks like this annotation was made on a document that is not publicly available.
To view it’s annotations, a document's address must start with HTTP or HTTPS."